packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -133,7 +150,7 @@ contract L1Block is ISemver, IGasToken {
     ///   7. _blobBaseFee        L1 blob base fee.
     ///   8. _hash               L1 blockhash.
     ///   9. _batcherHash        Versioned hash to authenticate batcher by.
-    function setL1BlockValuesEcotone() external {
+    function setL1BlockValuesEcotone() public {
```

Comment:
Changing it to `public` will have effects on prior ecotone usage, since the gas might change (although no explicit arguments to put in memory). I think the ecotone gas limits are conservative and have some buffer for changes like this, and we have tests, but it would still be nice to minimize changes.

Maybe instead instead we can change it to `internal`, keep an `external` function for ecotone that calls the `internal` one, and introduce the new isthmus one as `external` also? And document the calldata usage between these?

Reply:
To address this, we should follow [this](https://github.com/ethereum-optimism/design-docs/blob/main/smart-contract-feature-development.md) design doc and introduce a `L1BlockInterop is L1Block` contract